.github/workflows/deploy_staging.yml

```diff
@@ -0,0 +1,21 @@
+name: "Build Docker"
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  deploy-staging:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Generate firebase file
+        run: echo "${{secrets.FIREBASE_KEY}}" >> firebase.json
+
+      - name: Build the Docker image
+        run: docker build . --file Dockerfile --tag my-image-name:$(date +%s)
```

Comment:
You should setup your own docker instead of relying on the platform docker. This will work for the GitHub Ubuntu runners but might create an issue for the self hosted runners.

Use Qemu as well to allow the multi environment image creation in case we decide to use the Graviton

```
      -
        name: Set up QEMU
        uses: docker/setup-qemu-action@v3
      -
        name: Set up Docker Buildx
        uses: docker/setup-buildx-action@v3
```

Reply:
added